joomla/joomla-cms
# [com_joomlaupdate] Check for Self-Update Feature

Pull Request for Issue #11440
## Notice

Don't do that **never** on your Live Site please install a dev / test install!
#### Summary of Changes

This PR add the functionality that if a update for the update component is in place it should show the user to first update the com_joomlaupdate component.

With the new code in place we check exiplit on every call to the com_joomlaupdate default view for an update of the update component.
If there is a update we don't allow `Live Update`  but the `Upload & Install` still works as before!
If we have the last update installed -> everything works as before
#### Testing Instructions

Ok this comes a bit complicated :smile: (please follow exactly the steps)
- Install 3.6.0 https://github.com/joomla/joomla-cms/releases/3.6.0
- ignore all messages to update etc.
- go to the Backend -> Extensions -> Manage -> Update Sites
- disable the update site for com_joomlaupdate (Joomla! Update Component Update Site)
- install the patched  [com_joomlaupdate](http://www.jah-tz.de/downloads/download.php?file=com_joomlaupdate_UpdateMeFirst.zip&folder=core/extensions/) e.g. via install from URL (it cointains this PR and also a new update server!)
- double check there is a new Update server for com_joomlaupdate (!)
- go to com_joomlaupdate
- see the mesage that you can't use the `Live Update`
- follow the link (`Click here to update the component.`)
- install the update (it just move the version from 3.6.1 to 3.6.2 to pass the update check)
- install the Joomla Core Update
- confirm all works!
#### Why and how
###### Disable the official Update Site

If we not disable that it offers us the current com_joomlaupdate package which not include my changes.
##### Why a own Update server

Else i can't ship my changes to test this pull requrest as Update of the original com_joomlaupdate ;)
#### Documentation Changes Required

The Core Update Component now inlcues a function to call for a `Self Update`
#### Additional Infos

Please give feedback:
- On language strings
- Do we need both the `error message` and the changed Live Update View? Or should we remove the `error message`
- On code! It works but maybe it can be improved?! -> Feedback please!
- On everything else you notice we can improve!
#### Question

What do we with users that are unable to access com_installer but are allowed to update joomla? => Handel that with a language string? Please suggest one!

Great, you have read all and this this line too your are awesome :smile: Thanks!
